tests/test_core_error_handling.py

```
@@ -0,0 +1,416 @@
+            raise ValueError("Generic error")
+
+        with patch('core.error_handling.logger') as mock_logger:
+            result = failing_func()
+            assert result == "fallback"
+            mock_logger.error.assert_called_once()
+
+    def test_decorator_custom_severity(self):
+        """Test decorator with custom severity"""
+        @handle_component_error("test_component", severity=ErrorSeverity.CRITICAL)
+        def failing_func():
+            raise Exception("Critical error")
```

Ruff (0.14.10)

Create your own exception

(TRY002)